app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt

```diff
@@ -859,23 +895,27 @@ class LocationSensorManager : LocationSensorManagerBase() {
             if (lastUpdateLocation[serverId] == updateLocationString) {
                 if (now < (lastLocationSend[serverId] ?: 0) + 900000) {
                     Log.d(TAG, "Duplicate location received, not sending to HA")
+                    logLocationUpdate(location, updateLocation, serverId, trigger, LocationHistoryItemResult.SKIPPED_DUPLICATE)
                     return
                 }
             } else {
-                if (now < (lastLocationSend[serverId] ?: 0) + 5000 && !geofenceUpdate && !highAccuracyModeEnabled) {
+                if (now < (lastLocationSend[serverId] ?: 0) + 5000 && trigger?.isGeofence != true && !highAccuracyModeEnabled) {
                     Log.d(
                         TAG,
                         "New location update not possible within 5 seconds, not sending to HA"
                     )
+                    logLocationUpdate(location, updateLocation, serverId, trigger, LocationHistoryItemResult.SKIPPED_DEBOUNCE)
                     return
                 }
             }
         } else {
             Log.d(TAG, "Skipping location update due to old timestamp ${location.time} compared to $now")
+            logLocationUpdate(location, updateLocation, serverId, trigger, LocationHistoryItemResult.SKIPPED_OLD)
             return
         }
         lastLocationSend[serverId] = now
         lastUpdateLocation[serverId] = updateLocationString
+        logLocationUpdate(location, updateLocation, serverId, trigger, LocationHistoryItemResult.SENT)
```

Comment:
I wonder should we consider logging it here as sent or should we consider also logging when we fail to send the location update down below in the `try/catch`?

Reply:
Could be, but would it be very useful to know that the location was OK but it was unable to send it? For example depending on the device and server config the server being offline can produce different exceptions, so it's harder to make the error more specific without storing the raw exception (which I'd avoid).

Reply:
Well I think Sent or Failed to Send would indicate that its a valid report to be sent to the server and its either successful or not. I dont think we need to store the exception just the frequency of failures should be enough to go by.

Reply:
Updated, it also better matches log output.
![Screenshot showing a row with 'Failed to send' and a small icon of an alert sign](https://github.com/home-assistant/android/assets/8148535/02cac039-360f-4989-9540-c8e2e0421ed6)